src/pkg/services/m365/api/lists.go

```diff
@@ -658,6 +662,44 @@ func hasHyperLinkFields(additionalData map[string]any) (map[string]any, string,
 	return nil, "", false
 }
 
+func hasMetadataFields(additionalData map[string]any) ([]map[string]any, string, bool) {
+	for fieldName, value := range additionalData {
+		switch valType := reflect.TypeOf(value).Kind(); valType {
+		case reflect.Map:
+			metadataFields, areMetadataFields := getMetadataFields(value)
+			if areMetadataFields {
+				return []map[string]any{metadataFields}, fieldName, true
+			}
+
+		case reflect.Slice:
+			mmdfs := make([]map[string]any, 0)
+
+			multiMetadataFields, ok := value.([]any)
+			if !ok {
+				continue
+			}
+
+			for _, mdfs := range multiMetadataFields {
+				metadataFields, areMetadataFields := getMetadataFields(mdfs)
+				if areMetadataFields {
+					mmdfs = append(mmdfs, metadataFields)
+				}
+			}
+
+			if len(mmdfs) > 0 {
+				return mmdfs, fieldName, true
+			}
+		}
+	}
+
+	return nil, "", false
```

Comment:
would you need a boolean return if you returned an empty slice by default?  Or, alternatively, can the caller check for `len(fieldName) == 0` instead of the bool?